sdk/core/Azure.Core.TestFramework/src/ClientTestFixtureAttribute.cs

```diff
@@ -34,41 +54,93 @@ public IEnumerable<TestSuite> BuildFrom(ITypeInfo typeInfo, IPreFilter filter)
         {
             if (_serviceVersions.Any())
             {
-                foreach (object serviceVersion in _serviceVersions)
+                if (_additionalParameters?.Any() ?? false)
                 {
-                    var syncFixture = new TestFixtureAttribute(false, serviceVersion);
-                    var asyncFixture = new TestFixtureAttribute(true, serviceVersion);
-
-                    foreach (TestSuite testSuite in asyncFixture.BuildFrom(typeInfo, filter))
+                    foreach (var parameter in _additionalParameters)
                     {
-                        Process(testSuite, serviceVersion, true);
-                        yield return testSuite;
+                        foreach (object serviceVersion in _serviceVersions)
+                        {
+                            var syncFixture = new TestFixtureAttribute(false, serviceVersion, parameter);
```

Comment:
This ctor takes an array. Let's make only the array creation conditional instead of duplicating the loop.

Reply:
Thanks, I missed that - I really didn't like the duplication.